### ISSUE TITLE
Feature for encoding and decoding sex of samples/genomes in VCF header

### ISSUE THREAD

**Issue by wkretzsch** [samtools/bcftools#308]
I would like to encode the sex of the samples contained in a VCF in the VCF header. 

Based on the discussion at http://sourceforge.net/p/vcftools/mailman/message/28587354/, it appears that the best way of doing this in the absence of a ##GENOME field would be to encode sex using the ##SAMPLE field.  

So, what I'm envisioning are additions to `bcftools` that work in the following way:
````
bcftools reheader --add-sex sex_file file.vcf.gz
bcftools query --list-samples-with-sex > sex_file.2
````
, where `sex_file` and `sex_file.2` would not differ, and where `sex_file` would look like this:
````
sample sex 
samp1 female
samp2 male
samp3 male
...
````
Ideally, sex awareness would also be added to the `bcftools convert` tools that convert to and from sex-aware file formats (for example Impute2 format).

What do people think?

**Comment by adamspargo** [comment]
There is also the FMF used by bgt and others:

https://github.com/lh3/bgt#iphenotype

**Comment by wkretzsch** [comment]
Cool, thanks for the heads up.

**Comment by adamspargo** [comment]
I don't think this was the intension of the ##SAMPLE field, I'm assuming that I'll need to read sample files to get at further information about the samples in the VCF file, rows in sample files corresponding to columns in VCF files.

**Comment by mcshane** [comment]
This is a good idea. I'd like to make sure we agree with whatever the spec actually says. I kind of lost where the various ##SAMPLE and ##GENOME discussions ended up.

We'd then want to support adding this information from simple samples files, PED files and FMF files, etc... and, yes, using in the various convert options.

**Comment by wkretzsch** [comment]
The ##GENOME header is not part of the VCF 4.2 spec.  Genomes are only reference by the ##SAMPLE header.  I don't see a way of adding sample/genome meta information in the latest VCF spec, except for relationships between genomes in the ##PEDIGREE tag.  If we want this kind of meta information, then I think we'll need to extend the spec to include a ##GENOME header...

**Comment by pd3** [comment]
The sex can be added as an attribute to ##SAMPLE. The VCF specification lists reserved keys and examples, but new tags and attributes can be added when necessary:

     ##SAMPLE=<ID=Sample,Sex=Male,...>
